clang/lib/Sema/SemaDeclCXX.cpp

```diff
@@ -11322,9 +11322,22 @@ Decl *Sema::ActOnConversionDeclarator(CXXConversionDecl *Conversion) {
       << ClassType << ConvType;
   }
 
-  if (FunctionTemplateDecl *ConversionTemplate
-                                = Conversion->getDescribedFunctionTemplate())
+  if (FunctionTemplateDecl *ConversionTemplate =
+          Conversion->getDescribedFunctionTemplate()) {
+    if (ConvType->isUndeducedAutoType()) {
+      Diag(Conversion->getTypeSpecStartLoc(), diag::err_auto_not_allowed)
+          << Conversion->getTypeSourceInfo()
+                 ->getTypeLoc()
+                 .getContainedAutoTypeLoc()
```

Comment:
I think this is a slightly bigger hammer than it needs to be.
I think we should just use `->getTypeLoc().getSourceRange()` (yes, it would be the source range of the type + qualifiers, but i think that's as good, if not better

Reply:
I went this way mostly because I didn't want `const` qualifier on the member function to be highlighted, as it's not relevant. But it's getting a bit worse than this if we go your way:
```
    4 |   operator const auto() const { return 1.2; }     // error: conversion function template
      |                  ^~~~~~~~~~~~
    6 |   operator decltype(auto)() const { return 1.2; }     // error: conversion function template
      |            ^~~~~~~~~~~~~~~~~~~~~~
```

Reply:
Just the  return type.
There is even simpler actually

```cpp
getReturnTypeLoc(Conversion).getSourceRange()
```

Reply:
Nice! I updated the PR.